packages/spektate/src/Validation.ts

```diff
@@ -0,0 +1,349 @@
+import * as azure from "azure-storage";
+import {
+  fetchAuthor,
+  getDeployments,
+  getRepositoryFromURL,
+  IDeployment
+} from "./IDeployment";
+import { AzureDevOpsPipeline } from "./pipeline/AzureDevOpsPipeline";
+import { IAuthor } from "./repository";
+import {
+  getManifestSyncState as getADOClusterSync,
+  IAzureDevOpsRepo
+} from "./repository/IAzureDevOpsRepo";
+import {
+  getManifestSyncState as getGitHubClusterSync,
+  IGitHub
+} from "./repository/IGitHub";
+
+/**
+ * Validation error interface
+ */
+export interface IValidationError {
+  message: string;
+  code?: string;
+  stack?: string;
+}
+
+/**
+ * Errors interface
+ */
+export interface IErrors {
+  errors: IValidationError[];
+}
+
+/**
+ * Validates all configuration for a spektate instance
+ * @param storageAccountName storage account name
+ * @param storageAccountKey storage account key0
+ * @param storageTableName storage table name
+ * @param storagePartitionKey storage partition key
+ * @param orgName Azure DevOps organization name
+ * @param projectName Azure DevOps project name
+ * @param pipelinePAT Azure DevOps pipeline personal access token
+ * @param sourceRepoAccessToken source repo access token (same as pipeline token if it has access)
+ * @param manifestRepo manifest repo name (eg. test-manifests)
+ * @param manifestAccessToken manifest repo access token (same as pipeline token if it has access)
+ * @param manifestRepoGitHubOrgOrUsername GitHub organization or username for the manifest
+ * repository, eg. for github.com/microsoft/bedrock.git, this would be "microsoft"
+ */
+export const validateConfiguration = async (
+  storageAccountName: string,
+  storageAccountKey: string,
+  storageTableName: string,
+  storagePartitionKey: string,
+  orgName: string,
+  projectName: string,
+  pipelinePAT: string,
+  sourceRepoAccessToken: string,
+  manifestRepo: string,
+  manifestAccessToken: string,
+  manifestRepoGitHubOrgOrUsername?: string
+): Promise<IErrors> => {
```

Comment:
Since this has to do with top level validations, might want to wrap this all in a try/catch and ensure any errors thrown during any `awaits` are returned as instance of `IErrors`